File: ai-honeypot-fullstack/deploy/integrations/python/ingest_client.py
from __future__ import annotations

import os
from typing import Any, Dict, Optional

import requests


def build_ingest_url(base_url: str) -> str:
    base = (base_url or "").strip().rstrip("/")
    if not base:
        raise ValueError("Missing HONEYPOT_BASE_URL")
    if base.lower().endswith("/api"):
        return f"{base}/ingest"
    return f"{base}/api/ingest"


def send_honeypot_event(
    event: Dict[str, Any],
    base_url: Optional[str] = None,
    api_key: Optional[str] = None,
    timeout_sec: float = 6.0,
) -> Dict[str, Any]:
    base_url = base_url or os.getenv("HONEYPOT_BASE_URL", "")
    api_key = api_key or os.getenv("HONEYPOT_API_KEY", "")
    if not api_key:
        return {"ok": False, "status_code": 0, "data": None, "error": "Missing HONEYPOT_API_KEY"}

    try:
        url = build_ingest_url(base_url)
    except Exception as exc:
        return {"ok": False, "status_code": 0, "data": None, "error": str(exc)}

    try:
        resp = requests.post(
            url,
            json=event or {},
            headers={"X-API-Key": api_key, "Content-Type": "application/json"},
            timeout=timeout_sec,
        )
        try:
            payload = resp.json()
        except Exception:
            payload = {"raw": resp.text}
        return {"ok": resp.ok, "status_code": resp.status_code, "data": payload, "error": None}
    except requests.RequestException as exc:
        return {"ok": False, "status_code": 0, "data": None, "error": str(exc)}


if __name__ == "__main__":
    sample_event = {
        "event_type": "auth_fail",
        "url_path": "/login",
        "http_method": "POST",
        "captured_data": {"reason": "invalid_password"},
        "session_id": "py-smoke-local",
    }
    print(send_honeypot_event(sample_event))

File: ai-honeypot-fullstack/deploy/integrations/php/ingest_client.php
<?php
declare(strict_types=1);

function honeypot_build_ingest_url(string $baseUrl): string
{
    $base = rtrim(trim($baseUrl), "/");
    if ($base === "") {
        throw new InvalidArgumentException("Missing HONEYPOT_BASE_URL");
    }
    if (str_ends_with(strtolower($base), "/api")) {
        return $base . "/ingest";
    }
    return $base . "/api/ingest";
}

/**
 * Send one security event to the platform.
 * Returns:
 * [
 *   'ok' => bool,
 *   'status_code' => int,
 *   'data' => mixed,
 *   'error' => string|null
 * ]
 */
function honeypot_send_event(array $event, ?string $baseUrl = null, ?string $apiKey = null, int $timeoutSec = 6): array
{
    if (!function_exists("curl_init")) {
        return ["ok" => false, "status_code" => 0, "data" => null, "error" => "cURL extension not enabled"];
    }

    $baseUrl = $baseUrl ?? getenv("HONEYPOT_BASE_URL") ?: "";
    $apiKey = $apiKey ?? getenv("HONEYPOT_API_KEY") ?: "";

    if ($apiKey === "") {
        return ["ok" => false, "status_code" => 0, "data" => null, "error" => "Missing HONEYPOT_API_KEY"];
    }

    try {
        $url = honeypot_build_ingest_url($baseUrl);
    } catch (Throwable $e) {
        return ["ok" => false, "status_code" => 0, "data" => null, "error" => $e->getMessage()];
    }

    $ch = curl_init($url);
    $payload = json_encode($event, JSON_UNESCAPED_SLASHES);
    if ($payload === false) {
        return ["ok" => false, "status_code" => 0, "data" => null, "error" => "JSON encode failed"];
    }

    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_POST => true,
        CURLOPT_POSTFIELDS => $payload,
        CURLOPT_HTTPHEADER => [
            "Content-Type: application/json",
            "X-API-Key: " . $apiKey,
        ],
        CURLOPT_CONNECTTIMEOUT => $timeoutSec,
        CURLOPT_TIMEOUT => $timeoutSec,
    ]);

    $responseBody = curl_exec($ch);
    $statusCode = (int) curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $curlError = curl_error($ch);
    curl_close($ch);

    if ($responseBody === false) {
        return ["ok" => false, "status_code" => 0, "data" => null, "error" => $curlError ?: "Request failed"];
    }

    $decoded = json_decode($responseBody, true);
    $data = json_last_error() === JSON_ERROR_NONE ? $decoded : ["raw" => $responseBody];

    return [
        "ok" => $statusCode >= 200 && $statusCode < 300,
        "status_code" => $statusCode,
        "data" => $data,
        "error" => null,
    ];
}

/*
Example:
$result = honeypot_send_event([
    "event_type" => "auth_fail",
    "url_path" => "/login",
    "http_method" => "POST",
    "captured_data" => ["reason" => "invalid_password"],
    "session_id" => "php-login-" . time(),
]);
*/

